### PR TITLE
Delete v1beta3 package from api package and update code base accordingly

### PR DESCRIPTION
**Pull request by waodim**
## Description

NOTE: This is another PR from a series of PRs that deal with removing v1beta3 from operator version 1.3

This should be the last PR. 

- I removed v1beta3
- I changed storage version to v1beta2
- Fixed codebase

https://dt-rnd.atlassian.net/browse/K8S-11056

## How can this be tested?

Deploy operator, test it as good as you can

### CODE REVIEW

File: doc/coding-style-guide.md [hunk 1]
```diff
@@ -195,7 +195,6 @@ package abc
 
 import (
     dynakubev1beta2  "github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta2/dynakube"
-    dynakubev1beta3  "github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"
```
Comment (0sewa0) [inline doc/coding-style-guide.md:198]: here its just an example how you would import different versions of the same api

doesn't really matter, you could replace it with 
```
    dynakubev1beta1  "github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta1/dynakube"
    dynakubev1beta2  "github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta2/dynakube"
```

Comment (waodim) [reply doc/coding-style-guide.md:198]: done https://github.com/Dynatrace/dynatrace-operator/pull/3703/commits/0874d9c4a05488ef357ae4441e505dc134b1218d

**Comment by albertogdd** [comment]
Working on my cluster, but I triggered E2E to double check

**Review by albertogdd — APPROVED** [review]
So I see mainly istio tests fail (because its fix came after the release branch). It's a pass for me.

**Comment by andriisoldatenko** [comment]
seems okey:
image
LGTM, deployed to my cluster.

**Review by andriisoldatenko — APPROVED** [review]

**Review by wepudt — APPROVED** [review]
lgtm

**Pull request merged**